.github/pull_request_template.md

```diff
@@ -0,0 +1,53 @@
+Fixes #XXXX.
+
+(Replace XXXX with the id of the issue fixed in this PR. Remove this line if there is no corresponding
+issue. Don't reference the issue in the title of this pull-request.)
+
+Add tags to your PR if you are a committer (only committers have the right to add tags). Add [Design Review] tag
+if this PR should better be reviewed by at least two people.
+Don't forget to add the following tags (if applicable): [Incompatible], [Release Notes], [Compatibility], [Security],
+[Development Blocker]. Add at least one [Area - ] tag, consider creating a new one if none of the existing [Area - ]
+tags is applicable.
+
+### Description
+
+Describe the goal of this PR, what problem are you fixing. If there is a corresponding issue (referenced above), it's
+not necessary to repeat the description here, however, you may choose to keep one summary sentence.
+
+Describe your patch: what did you change in code? How did you fix the problem?
+
+If there are several relatively logically separate changes in this PR, list them. For example:
+ - Fixed the bug ...
+ - Renamed the class ...
+ - Added a forbidden-apis entry ...
+
+Some of the aspects mentioned above may be omitted for simple and small PRs.
+
+### Design
+
+Please describe any design decisions made, including:
+ - Choice of algorithms
+ - Behavioral aspects. What configuration values are acceptable? How are corner cases and error conditions handled, such
+   as when insufficient resources are available?
+ - Class organization and design (how the logic is split between classes, inheritance, composition, design patterns)
+ - Method organization and design (how the logic is split between methods, parameters and return types)
+ - Naming (class, method, API, configuration, HTTP endpoint, names of emitted metrics)
+
+In addition, describe _at least one_ alternative design (or mention alternative name) for every design (or naming)
+decision point and compare the alternatives with the designs that you've implemented (or the names you've chosen).
+
+If you already did this in the associated issue (e. g. a "Proposal" issue), leave the following sentence:
+
+Design of this change is discussed [here](<link to Github issue or comment where you discuss the design>).
+
+This section may be omitted for really simple and small patches. However, any patch that adds a new production class
+almost certainly shouldn't omit this section.
+
+<hr>
+
+I've self-reviewed this PR (including using the [concurrency checklist](
+https://github.com/leventov/java-concurrency-checklist)).
```

Comment:
I wonder if this should maybe instead be a wiki page in this repo, or some other official ASF resource?

Reply:
I don't like this personal association myself, because I think it diminishes chances that people will use the checklist. I've moved it to https://github.com/code-review-checklists/java-concurrency.

Reply:
IMO it's better in the main Druid repo, which would also help generate more discussion on the contents of the doc (there hasn't been much on the dev list from what I can see).